processing/src/main/java/org/apache/druid/query/aggregation/first/StringAggregatorUtils.java

```diff
@@ -87,7 +87,7 @@ public static void writePair(
 
     if (pair.rhs != null) {
       mutationBuffer.position(position + Long.BYTES + Integer.BYTES);
-      mutationBuffer.limit(maxStringBytes);
+      mutationBuffer.limit(position + Long.BYTES + Integer.BYTES + maxStringBytes);
```

Comment:
Why was this change needed?

Reply:
This is to fix an existing bug i found. The limit method for buffer sets it at absolute position. The correct limit should be maxStringBytes after where we are writing the String to the buffer (which is position + Long.BYTES + Integer.BYTES). Hence, limit should be position + Long.BYTES + Integer.BYTES + maxStringBytes

Reply:
Removed this. Will fix as part of separate PR